pkg/nsx/services/common/types.go

```diff
@@ -15,30 +15,39 @@ import (
 )
 
 const (
-	HashLength                   int    = 8
-	MaxTagLength                 int    = 256
-	TagScopeCluster              string = "nsx-op/cluster"
-	TagScopeNamespace            string = "nsx-op/namespace"
-	TagScopeSecurityPolicyCRName string = "nsx-op/security_policy_cr_name"
-	TagScopeSecurityPolicyCRUID  string = "nsx-op/security_policy_cr_uid"
-	TagScopeRuleID               string = "nsx-op/rule_id"
-	TagScopeGroupType            string = "nsx-op/group_type"
-	TagScopeSelectorHash         string = "nsx-op/selector_hash"
-	TagScopeNCPCluster           string = "ncp/cluster"
-	TagScopeNCPProject           string = "ncp/project"
-	TagScopeNCPVIFProject        string = "ncp/vif_project"
-	TagScopeNCPPod               string = "ncp/pod"
-	TagScopeNCPVNETInterface     string = "ncp/vnet_interface"
+	HashLength                      int    = 8
+	MaxTagLength                    int    = 256
+	MaxIdLength                     int    = 255
+	TagScopeCluster                 string = "nsx-op/cluster"
+	TagScopeNamespace               string = "nsx-op/namespace"
+	TagScopeSecurityPolicyCRName    string = "nsx-op/security_policy_cr_name"
+	TagScopeSecurityPolicyCRUID     string = "nsx-op/security_policy_cr_uid"
+	TagScopeRuleID                  string = "nsx-op/rule_id"
+	TagScopeGroupType               string = "nsx-op/group_type"
+	TagScopeSelectorHash            string = "nsx-op/selector_hash"
+	TagScopeNSXServiceAccountCRName string = "nsx-op/nsx_service_account_name"
+	TagScopeNSXServiceAccountCRUID  string = "nsx-op/nsx_service_account_uid"
+	TagScopeNCPCluster              string = "ncp/cluster"
+	TagScopeNCPProject              string = "ncp/project"
+	TagScopeNCPVIFProject           string = "ncp/vif_project"
+	TagScopeNCPPod                  string = "ncp/pod"
+	TagScopeNCPVNETInterface        string = "ncp/vnet_interface"
 
 	GCInterval    = 60 * time.Second
 	FinalizerName = "securitypolicy.nsx.vmware.com/finalizer"
+
+	NSXServiceAccountFinalizerName = "nsxserviceaccount.nsx.vmware.com/finalizer"
 )
 
 var (
 	ResourceType               = "resource_type"
 	ResourceTypeSecurityPolicy = "SecurityPolicy"
 	ResourceTypeGroup          = "Group"
 	ResourceTypeRule           = "Rule"
+	// Used by NSXServiceAccountController
```

Comment:
This format is better
ResourceTypeClusterControlPlane = "clustercontrolplane" // Used by NSXServiceAccountController
Otherwise 
<img width="919" alt="image" src="https://user-images.githubusercontent.com/4385934/203684429-ebf5d2fc-b5da-4b64-b739-644e715698f9.png">

Reply:
Refactored here.